### PR TITLE
MYSQL80 TSQL updates

### PR DESCRIPTION
**Pull request by Cataldo71**
I was getting errors during startup due to use of tsql keywords in the SQL queries. the table name 'groups' and the field 'rank' were both flagging as invalid sql and throwing errors. I simply scoped those to the db/table so that they would resolve properly.

### CODE REVIEW

**Comment by Fabi** [comment]
you probably want to use  \`name` instead

**Comment by Cataldo71** [comment]
I tried that with both single/double quotes and tsql didn't seem to like it. technically i should be surrounding the col names with [ ] so maybe ill do that instead, but that doesnt seem to work with the groups table. I cannot find any reference to 'groups' being a protected keyword, but it seems that mysql 8 wants it scoped.

**Comment by Fabi** [comment]
But TSQL is a microsoft thing and not mysql?! I don't think mysql can handle [] at all

**Comment by Cataldo71** [comment]
sorry, i typed tsql meaning mysql.... I tried [ ] and you are correct, it is not recognized by mysql. Scoping the column to the table works just fine as does single quotes.

File: src/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -3877,7 +3877,7 @@ void ObjectMgr::LoadGroups()
                  "(SELECT COUNT(*) FROM character_instance WHERE guid = group_instance.leaderGuid AND instance = group_instance.instance AND permanent = 1 LIMIT 1), "
                  // 7              8
                  " groups.groupId, instance.encountersMask "
-                 "FROM group_instance LEFT JOIN instance ON instance = id LEFT JOIN groups ON groups.leaderGUID = group_instance.leaderGUID ORDER BY leaderGuid"
+                 "FROM group_instance LEFT JOIN instance ON instance = id LEFT JOIN wotlkcharacters.groups ON groups.leaderGUID = group_instance.leaderGUID ORDER BY leaderGuid"
```
Comment (Dramacydal) [inline src/game/Globals/ObjectMgr.cpp]: i guess wotlkcharacters is your local database name, i can be different for the others

**Comment by cyberium** [comment]
Something have to be done for mysql8 but not what is done here

**Comment by Phatcat** [comment]
something was done; https://github.com/cmangos/mangos-wotlk/commit/6a86a0be331f2e42253bdf52a37caf5bd06a4d01 - looks like I missed `` around groups in src/game/Maps/MapPersistentStateMgr.cpp

**Pull request closed without merging**